lightning/src/util/wakers.rs

```diff
@@ -340,6 +344,47 @@ impl Sleeper {
 	}
 }
 
+pub struct Sleep {
```

Comment:
There is already `Sleeper` in the repository, but `from_single_future` takes a future and isn't just a sleep.